frontend/public/module/k8s/k8s-models.ts

```diff
@@ -72,3 +80,53 @@ export const modelFor = (ref: K8sResourceKindReference) => {
  * NOTE: This will not work for CRDs defined at runtime, use `connectToModels` instead.
  */
 export const allModels = () => k8sModels;
+
+/**
+ * Use this hook to find the model for resources using only group and plural/kind
+ */
+export const useModelFinder = () => {
+  const models: ImmutableMap<string, K8sKind[]> = useSelector(({ k8s }) =>
+    k8s.getIn(['RESOURCES', 'models']),
+  );
+  const pluralsToModelMap = _.values(models.toJS()).reduce((acc, curr) => {
+    const ref = referenceForGroupVersionKind(curr.apiGroup)(curr.apiVersion)(curr.plural);
```

Comment:
This is going to get really confusing because we're blurring the distinction between `plural` and `kind`. You shouldn't call `referenceForGroupVersionKind` and pass something other than `kind` as the third value.

Reply:
Creating a new helper called `referenceForGroupVersionPlural`

Reply:
> Creating a new helper called `referenceForGroupVersionPlural`

I wouldn't create a general. This is very specific to this one function. Just have a helper scoped to this function for it.